pkg/connection/connection.go

```diff
@@ -196,9 +196,14 @@ func (c *Connection) exec(ctx context.Context, query string, args []driver.Value
 			return nil, err
 		}
 
-		defer importStatement.Close()
 		query = importStatement.GetUpdatedQuery()
-		errs.Go(func() error { return importStatement.UploadFiles(errctx) })
+		go func() {
+			defer importStatement.Close()
```

Comment:
Maybe a comment on why we need to close the import statement here and why we can't run this goroutine in the error group might be advisable.